boards/DIM/Src/App/states/App_DriveState.c

```diff
@@ -190,3 +199,27 @@ const struct State *App_GetDriveState(void)
 
     return &drive_state;
 }
+
+bool has_bms_fault(const struct DimCanRxInterface *can_rx)
+{
+    return App_CanRx_BMS_FAULTS_GetSignal_CHARGER_DISCONNECTED_IN_CHARGE_STATE(can_rx) ||
+           App_CanRx_BMS_FAULTS_GetSignal_CELL_UNDERVOLTAGE_FAULT(can_rx) ||
+           App_CanRx_BMS_FAULTS_GetSignal_MODULE_COMM_ERROR(can_rx) ||
+           App_CanRx_BMS_FAULTS_GetSignal_CELL_OVERVOLTAGE_FAULT(can_rx) ||
+           App_CanRx_BMS_FAULTS_GetSignal_CELL_UNDERTEMP_FAULT(can_rx) ||
+           App_CanRx_BMS_FAULTS_GetSignal_CELL_OVERTEMP_FAULT(can_rx) ||
+           App_CanRx_BMS_FAULTS_GetSignal_CHARGER_FAULT_DETECTED(can_rx) ||
+           App_CanRx_BMS_FAULTS_GetSignal_HAS_REACHED_MAX_V(can_rx) ||
+           App_CanRx_BMS_FAULTS_GetSignal_CHARGING_EXT_SHUTDOWN_OCCURRED(can_rx) ||
+           App_CanRx_BMS_FAULTS_GetSignal_TS_OVERCURRENT_FAULT(can_rx) ||
+           App_CanRx_BMS_FAULTS_GetSignal_PRECHARGE_ERROR(can_rx);
+}
+
+bool has_bms_warning(const struct DimCanRxInterface *can_rx)
+{
+    return App_CanRx_BMS_NON_CRITICAL_ERRORS_GetSignal_STACK_WATERMARK_ABOVE_THRESHOLD_TASK1_HZ(can_rx) ||
+           App_CanRx_BMS_NON_CRITICAL_ERRORS_GetSignal_STACK_WATERMARK_ABOVE_THRESHOLD_TASK1_KHZ(can_rx) ||
+           App_CanRx_BMS_NON_CRITICAL_ERRORS_GetSignal_STACK_WATERMARK_ABOVE_THRESHOLD_TASKCANRX(can_rx) ||
+           App_CanRx_BMS_NON_CRITICAL_ERRORS_GetSignal_STACK_WATERMARK_ABOVE_THRESHOLD_TASKCANTX(can_rx) ||
+           App_CanRx_BMS_NON_CRITICAL_ERRORS_GetSignal_WATCHDOG_TIMEOUT(can_rx);
+}
```

Comment:
Hmm I thought the decision was to stop checking individual errors like is done here, previously done by the error table. And instead check if the BMS is in its "FAULT" state.

Reply:
That’s a solid point. I don’t know why that slipped my mind I’ll switch to that.

Reply:
Can you think of any way to implement a cleaner check for warnings however? warnings are not currently state-dependent.